dotCMS/src/integration-test/java/com/dotcms/security/secret/SecretsStoreKeyStoreImplTest.java

```diff
@@ -0,0 +1,215 @@
+package com.dotcms.security.secret;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.util.UUIDGenerator;
+import java.util.Collection;
+import java.util.Optional;
+import org.apache.commons.lang.RandomStringUtils;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class SecretsStoreKeyStoreImplTest {
+
+    private final String LONG_KEY_NAME = new String(
+            "A"
+                    + "\u00ea"
+                    + "\u00f1"
+                    + "\u00fc"
+                    + "C"
+                    + "ASDSADFDDSFasfddsadasdsadsadasdq3r4efwqrrqwerqewrqewrqreqwreqwrqewrqwerqewrqewrqwerqwerqwerqewr43545324542354235243524354325423qwerewds fds fds gf eqgq ewg qeg qe wg egw    ww  eeR ASdsadsadadsadsadsadaqrewq43223t14@#$#@^%$%&%#$sfwf erqwfewfqewfgqewdsfqewtr243fq43f4q444fa4ferfrearge");
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+        IntegrationTestInitService.getInstance().init();
+        SecretsStore.INSTANCE.get().deleteAll();
+    }
+
+    /**
+     * This tests to insure that we have a secrets singleton
+     */
+    @Test
+    public void Test_Secrets_Singleton() throws Exception {
+
+        final SecretsStore secretsStore = SecretsStore.INSTANCE.get();
+
+        final SecretsStore secretsStore1 = SecretsStore.INSTANCE.get();
+
+        assertEquals("we have a SecretsStore singleton", secretsStore, secretsStore1);
+    }
+
+
+    /**
+     * This tests storing an retrieving a value from the SecretsStore
+     */
+    @Test
+    public void Test_Storing_a_Value() throws Exception {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)